### PR TITLE
Enrich spring boms changelog to include dependency upgrades

### PR DESCRIPTION
**Pull request by saragluna**
We expect Azure Spring users to use azure-spring-boot-bom and azure-spring-cloud-dependencies to manage their Azure Spring dependencies. This PR adds a `Dependency Upgrades` section under each release in CHANGELOG.

### CODE REVIEW

File: sdk/boms/azure-spring-boot-bom/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,73 @@
...
+- Upgrade to [Azure Identity 1.2.5](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/identity/azure-identity/CHANGELOG.md#release-history).
+- Upgrade to [Azure Key Vault Secret 4.2.7](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/keyvault/azure-security-keyvault-secrets/CHANGELOG.md#release-history).
+- Upgrade to [Azure MSAL 1.9.1](https://github.com/AzureAD/microsoft-authentication-library-for-java/releases/tag/v1.9.1).
+- Upgrade to [Azure Service Bus JMS 0.0.7](https://search.maven.org/artifact/com.microsoft.azure/azure-servicebus-jms/0.0.7/jar).
+- Upgrade to [Azure Spring Data Cosmos 3.6.0](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/cosmos/azure-spring-data-cosmos/CHANGELOG.md#360-2021-04-06).
+- Upgrade to [Azure Storage Blob 12.10.2](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/storage/azure-storage-blob/CHANGELOG.md#12102-2021-03-26).
+- Upgrade to [Azure Storage File Share 12.8.0](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/storage/azure-storage-file-share/CHANGELOG.md#1280-2021-01-14).
```
Comment (chenrujun) [inline sdk/boms/azure-spring-boot-bom/CHANGELOG.md:49]: This entry already exist in 3.3.0. maybe we should delete it in 3.4.0?

File: sdk/boms/azure-spring-boot-bom/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,73 @@
...
 
 ## 3.6.0 (2021-06-23)
-
+### Dependency Upgrades
+- Upgrade to [spring-boot-dependencies:2.5.0](https://repo.maven.apache.org/maven2/org/springframework/boot/spring-boot-dependencies/2.5.0/spring-boot-dependencies-2.5.0.pom).
+- Upgrade to [Azure Core 1.17.0](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/CHANGELOG.md#1170-2021-06-07).
+- Upgrade to [Azure Identity 1.3.1](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/identity/azure-identity/CHANGELOG.md#131-2021-06-08).
+- Upgrade to [Azure Key Vault Secret 4.3.0](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/keyvault/azure-security-keyvault-secrets/CHANGELOG.md#430-2021-06-17).
+- Upgrade to [Azure MSAL 1.10.1](https://github.com/AzureAD/microsoft-authentication-library-for-java/releases/tag/v1.10.1).
+- Upgrade to [Azure Service Bus JMS 0.0.7](https://search.maven.org/artifact/com.microsoft.azure/azure-servicebus-jms/0.0.7/jar).
```
Comment (chenrujun) [inline sdk/boms/azure-spring-boot-bom/CHANGELOG.md:10]: Same for this one. 

And if you plan to delete this, please check all other entries.

File: sdk/boms/azure-spring-boot-bom/CHANGELOG.md (hunk 1 above)
Comment (saragluna) [reply sdk/boms/azure-spring-boot-bom/CHANGELOG.md:49]: Just add this to show a full picture of which dependencies are included.

Comment (chenrujun) [reply sdk/boms/azure-spring-boot-bom/CHANGELOG.md:49]: Then it's not change log, it's just a duplication of the content of the pom. 
How about do like this:
1. Rename `### Dependency Upgrades` to `### Dependency Versions`
2. Convert the content to a table.

Comment (saragluna) [reply sdk/boms/azure-spring-boot-bom/CHANGELOG.md:49]: I'd prefer keeping the current format, by following https://github.com/spring-projects/spring-boot/releases.

Comment (saragluna) [reply sdk/boms/azure-spring-boot-bom/CHANGELOG.md:49]: Indeed it's not an upgrade, I can remove them。

Comment (chenrujun) [reply sdk/boms/azure-spring-boot-bom/CHANGELOG.md:49]: And I prefer to move the table to readme. 

The table should be like this:
|    | artifact-1 |artifcat-2|
| --| -- | --| 
|  bom:3.6.0| version-1-1| version-2-1| 
|  bom:3.5.0| version-1-2| version-2-1|

Comment (chenrujun) [reply sdk/boms/azure-spring-boot-bom/CHANGELOG.md:49]: And link for each version is not necessary. It's OK to add link to each artifact.

Comment (chenrujun) [reply sdk/boms/azure-spring-boot-bom/CHANGELOG.md]: So you deleted the duplicated items. It's OK for me.

**Review by chenrujun — APPROVED** [review]
LGTM.

**Pull request merged**